### PR TITLE
Add e2e-kind.sh

### PR DESCRIPTION
**Pull request by yaoxiaoqi**
# Changes

This PR adds a script which runs e2e tests locally by using kind. Depends on plumbing PR https://github.com/tektoncd/plumbing/pull/639.

/kind feature

-->

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
Add a script which runs e2e tests locally by using kind
```

### CODE REVIEW

File: test/e2e-kind.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+install_pipeline_crd
+
+failed=0
+
+# Run the integration tests
+header "Running Go e2e tests"
+go_test_e2e -timeout=40m ./test/... -run="TestTaskRunPipelineRunCancel|TestDAGPipelineRun" || failed=1
+
+# Run these _after_ the integration tests b/c they don't quite work all the way
```
Comment (ImJasonH) [inline test/e2e-kind.sh:36]: Should we add a `(( failed )) && fail_test` here too, so we fail fast if the first batch of tests fail, and don't bother to run the rest?

Comment (yaoxiaoqi) [reply test/e2e-kind.sh:36]: Sure. Should we add `(( failed )) && fail_test` to e2e-tests.sh too? I think it also needs this.

Comment (ImJasonH) [reply test/e2e-kind.sh:37]: Oh I didn't realize this was copied from e2e-tests.sh. In that case keeping the logic the same seems good, and we can open another PR to update both of them to fail faster, and discuss whether we want that.

**Pull request closed without merging**